src/main/java/network/brightspots/rcv/TabulatorSession.java

```diff
@@ -64,38 +64,38 @@
     timestampString = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss").format(new Date());
   }
 
-  // Visible for testing
-  @SuppressWarnings("unused")
-  String getOutputPath() {
-    return outputPath;
-  }
-
-  // Visible for testing
-  @SuppressWarnings("unused")
-  String getTimestampString() {
-    return timestampString;
-  }
-
-  // given a dominion style csv json file path:
+  // given a dominion style folder path:
   // read associated manifest data
   // read Dominion cvr json into CastVoteRecords
   // write CastVoteRecords to generic cvr csv files: one per contest
-  // return true if files are successfully written
-  // return false if an error is encountered
-  static boolean convertDominionCvrJsonToGenericCsv(String dominionCvrJsonFile) {
-    boolean success = true;
-    DominionCvrReader dominionCvrReader = new DominionCvrReader(dominionCvrJsonFile);
+  // return list of files written
+  static List<String> convertDominionCvrJsonToGenericCsv(String dominionDataFolder) {
+    DominionCvrReader dominionCvrReader = new DominionCvrReader(dominionDataFolder);
     List<CastVoteRecord> castVoteRecords = new ArrayList<>();
+    List<String> filesWritten;
     try {
       dominionCvrReader.readCastVoteRecords(castVoteRecords);
       ResultsWriter writer = new ResultsWriter();
-      writer.writeGenericCvrCsv(castVoteRecords, dominionCvrReader.getContests().values(),
-          dominionCvrJsonFile);
+      filesWritten = writer
+          .writeGenericCvrCsv(castVoteRecords, dominionCvrReader.getContests().values(),
+              dominionDataFolder);
     } catch (Exception exception) {
       Logger.log(Level.SEVERE, "Failed to convert Dominion CVR to CSV:\n%s", exception.toString());
-      success = false;
+      filesWritten = null;
```

Comment:
Is it possible that some files were written even if there was a failure for one of them though?

Reply:
Yeah, the idea is a null return value indicates failure in case the caller cares.  More important to fail loudly here (and pretty much everywhere in the Tabulator) than try to recover some partial results and do something with them.  I'll indicate that in the comment.